code/game/objects/items/devices/transfer_valve.dm

```diff
@@ -166,12 +169,21 @@
 	underlays.Cut()
 	if(!tank_one && !tank_two && !attached_device)
 		return
+
 	if(tank_one)
-		. += "[tank_one.icon_state]"
+		var/tank_one_icon_state = tank_one.icon_state
+		if(!(tank_one_icon_state in TTV_TANK_ICON_STATES)) //if no valid sprite fall back to an oxygen tank
+			tank_one_icon_state = "oxygen"
```

Comment:
please put a stack_trace in here, we want this to runtime if there's icon fuckery afoot